spec/solr_indexer_spec.rb

```diff
@@ -1,74 +1,90 @@
+    let(:document) { JSON.load_file('spec/fixture/jsonfile/berkeley_public_pdf.json') }
 
-    before do
-      allow(solr).to receive(:update)
-      allow(RSolr).to receive(:connect).and_return(solr)
-      solr_indexer.update(file_path)
+    it 'does nothing if reference_urls are nil' do
+      indexer = Gingr::SolrIndexer.new
+      expect { indexer.update_reference_urls! document }.not_to change { document }
     end
 
-    context 'update reference urls' do
-      let(:solr_indexer) { described_class.new(url, reference_urls) }
-      let(:reference_urls) do
-        { 'geoserver_secure' => 'http://fake_geoserver_secure:8081',
-          'geoserver' => 'http://fake_geoserver:8080',
-          'download' => 'https://fake_spatial.lib.berkeley.edu' }
-      end
+    it 'updates references if configured to do so' do
+      refs = { geoserver_url: 'http://geoserver-at-init/' }
+      indexer = Gingr::SolrIndexer.new(nil, refs)
+      expect { indexer.update_reference_urls! document }.to change { document }
+      expect(document['dct_references_s']).to match 'http://geoserver-at-init/'
+    end
+  end
 
-      it 'should call solr' do
-        expect(solr).to have_received(:update).with(
-          params: { commitWithin: 5000, overwrite: true },
-          data: [[doc].flatten[0]].to_json,
-          headers: { 'Content-Type' => 'application/json' }
-        )
-      end
+  describe '#index_directory' do
+    it 'adds all .json files to solr' do
+      files = ['foo.xml', 'bar.json', 'baz.json'].shuffle
+      expect(Find).to receive(:find).with('directory').and_return files
+      Gingr::SolrIndexer.any_instance.stub(:add)
 
-      it 'should call the update reference field method' do
-        expect(solr_indexer.update_reference_urls?).to eq(true)
-      end
+      indexer = Gingr::SolrIndexer.new
+      indexer.index_directory('directory')
+      expect(indexer).to have_received(:add).with('bar.json')
+      expect(indexer).to have_received(:add).with('baz.json')
+      expect(indexer).not_to have_received(:add).with('foo.xml')
     end
+  end
 
-    context 'not update reference urls' do
-      let(:solr_indexer) { described_class.new(url) }
-      it 'should not call the update reference field method' do
-        solr_indexer.update(file_path)
-        expect(solr_indexer.update_reference_urls?).to eq(false)
-      end
+  describe '#add' do
+    let(:document) { JSON.load_file document_path }
+    let(:document_path) { 'spec/fixture/jsonfile/berkeley_public_pdf.json' }
+
+    it 'passes documents to the rsolr client' do
+      solr = spy(RSolr::Client)
+      indexer = Gingr::SolrIndexer.new(solr)
+      indexer.add(document)
+      expect(solr).to have_received(:add).with(document)
     end
-  end
 
-  describe '#commit' do
-    before do
-      allow(RSolr).to receive(:connect).and_return(solr)
-      allow(solr).to receive(:commit)
+    it 'automatically loads filepaths as JSON' do
+      solr = spy(RSolr::Client)
```

Comment:
I discovered `spy` as an alternative to `instance_double`. This allows placing expectations after the code that calls them, making the test more readable. `have_received` serves a similar purpose.

Reply:
ok, good to know

Reply:
Will try spy next time